android/src/main/kotlin/com/signify/hue/flutterreactiveble/channelhandlers/ScanDevicesHandler.kt

```diff
@@ -33,19 +33,21 @@ class ScanDevicesHandler(private val bleClient: com.signify.hue.flutterreactiveb
     }
 
     private fun startDeviceScan() {
-            scanParameters?.let { params ->
-                scanForDevicesDisposable = bleClient.scanForDevices(params.filter, params.mode)
-                        .observeOn(AndroidSchedulers.mainThread())
-                        .subscribe(
-                        { scanResult ->
-                            handleDeviceScanResult(converter.convertScanInfo(scanResult))
-                        },
-                        { throwable ->
-                            Timber.d("Error while scanning for devices:  ${throwable.message}")
-                            handleDeviceScanResult(converter.convertScanErrorInfo(throwable.message))
-                        }
-                )
-            } ?: handleDeviceScanResult(converter.convertScanErrorInfo("Scanparameters are not being set"))
+        scanParameters?.let { params ->
+            scanForDevicesDisposable = bleClient.scanForDevices(params.filter, params.mode, params.locationServiceIsMandatory)
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(
+                            { scanResult ->
+                                handleDeviceScanResult(converter.convertScanInfo(scanResult))
+                            },
+                            { throwable ->
+
+                                Timber.d("Error while scanning for devices:  ${throwable.message}")
+                                handleDeviceScanResult(converter.convertScanErrorInfo(throwable.message))
+                            }
+                    )
+        }
+                ?: handleDeviceScanResult(converter.convertScanErrorInfo("Scanparameters are not being set"))
```

Comment:
What's `Scanparameters`? Casing doesn't much the variable name and neither it is natural language. I'd suggest to use natural language; if you don't find this a good idea, then match the name of the variable exactly, otherwise it's confusing.